app/code/core/Mage/Sitemap/Model/Sitemap.php

```diff
@@ -195,13 +195,16 @@ public function generateXml()
         /**
          * Generate cms pages sitemap
          */
+        $homepage = (string)Mage::getStoreConfig('web/default/cms_home_page', $storeId);
         $changefreq = (string)Mage::getStoreConfig('sitemap/page/changefreq', $storeId);
         $priority   = (string)Mage::getStoreConfig('sitemap/page/priority', $storeId);
         $collection = Mage::getResourceModel('sitemap/cms_page')->getCollection($storeId);
         foreach ($collection as $item) {
+            $url = $item->getUrl();
+			if ( $url == $homepage) { $url = ''; $priority = 1; }
```

Comment:
And if someone responds about the prio autosetting it to 1 (who know a domain "could" have an main page that is not the most important) ... I understand. But I do think this is best. 

It has always bugged me that not the real domain, but rather the CMS handle was sent to Google (or Bing for that matter) .... then we block it via robots or in nginx create a 301 for the CMS handle .... simplest solution is to: just send the main URL